main.py

```diff
@@ -690,19 +728,82 @@ def update_port_status(self, port_status, source):
             interface = source.switch.get_interface_by_port_no(port_no)
             interface.deactivate()
 
+        self._intf_state_seen_num[switch.id][interface.id] = xid_seq_num
         event_name += status
         content = {'interface': interface}
 
         event = KytosEvent(name=event_name, content=content)
         self.controller.buffers.app.put(event)
 
-        # pylint: disable=protected-access
-        state_desc = {v: k for k, v in PortState._enum.items()}
-        # pylint: enable=protected-access
         state = state_desc.get(port.state.value, port.state.value)
         msg = 'PortStatus %s interface %s:%s state %s'
         log.info(msg, status, source.switch.id, port_no, state)
 
+    async def _handle_port_desc(self, switch, reply):
+        """Update interfaces on switch based on port_list information."""
+        port_list = reply.body
+        interfaces = []
+        for port in port_list:
+            config = port.config
+            if (port.supported == 0 and
+                    port.curr_speed.value == 0 and
+                    port.max_speed.value == 0):
+                config = PortConfig.OFPPC_NO_FWD
+
+            port_no = port.port_no.value
+
+            intf = switch.get_interface_by_port_no(port_no)
+            xid_seq_num = self._xid_seq_num[switch.id][int(reply.header.xid)]
+            if (
+                settings.SKIP_INTF_STATE_LATE_UPDATES and
+                intf and
+                xid_seq_num < self._intf_state_seen_num[switch.id][intf.id]
+            ):
+                # pylint: disable=protected-access
+                state_desc = {v: k for k, v in PortState._enum.items()}
+                # pylint: enable=protected-access
+                state = state_desc.get(port.state.value, port.state.value)
+                last_seen = self._intf_state_seen_num[switch.id][intf.id]
+                log.info(
```

Comment:
@italovalcy could you please try out this branch and run it a few times with the SDX test case that was catching it? We can also add a new e2e test with a trimmed down scenario if you can also point out an equivalent scenario.

Reply:
I added this `log.info` when skipping the message to be useful for us if we need to debug this in the future. Also, it can be handy as well for you to spot this message in the test case where it's expected to skip the port desc. I don't expect this message to be too noise, so it should be fine.

Reply:
Excellent. Let's ship it then.